docs/swarm/networking.md

```diff
@@ -0,0 +1,291 @@
+```
+
+The service VIP means you don't need to expose service-specific ports to make
+the service available to other services on the same overlay network. The swarm's
+internal load balancer automatically distributes requests to the service VIP
+among the active tasks.
+
+For example, you can add a `busybox` service on the same network as the `nginx`
+service. The busybox service can access `nginx` using the DNS name `my_web`:
+
+1. From a manager node, deploy a busybox service to the same network as
+`my_web`:
+
+    ```bash
+    $ docker service create \
+      --name my_busybox \
+      --network my-network \
+      busybox \
+      sleep 3000
+    ```
+
+2. Lookup the node where `my_busybox` is running:
+
+    ```bash
+    $ docker service ps my_busybox
+
+    ID                         NAME          IMAGE    NODE   DESIRED STATE  CURRENT STATE          ERROR
+    1dok2cmx2mln5hbqve8ilnair  my_busybox.1  busybox  node1  Running        Running 5 seconds ago
+    ```
+
+3. From the node where the busybox task is running, open an interactive shell to
+the busybox conatiner:
+
+    ```bash
+    $ docker exec -it my_busybox.1.1dok2cmx2mln5hbqve8ilnair /bin/sh
+    ```
+
+    You can deduce the container name as `<TASK-NAME>`+`<ID>`. Alternatively,
+    you can run `docker ps` on the node where the task is running.
+
+4. Query the DNS to view the VIP for the `my_web` service:
+
+    ```bash
+    # From inside the busybox container:
+    $ nslookup my_web
+
+    Server:    127.0.0.11
+    Address 1: 127.0.0.11
+
+    Name:      my_web
+    Address 1: 10.0.9.2 ip-10-0-9-2.us-west-2.compute.internal
+    ```
+
+    >**Note:** the examples here use `nslookup`, but you can use `dig` or any
+    available DNS query tool.
+
+5. Query the DNS to find the IP addresses of all the containers for the `my_web`
+service:
+
+    ```bash
+    # From inside the busybox container:
+    $ nslookup tasks.my_web
+
+    Server:    127.0.0.11
+    Address 1: 127.0.0.11
+
+    Name:      tasks.my_web
+    Address 1: 10.0.9.4 my_web.2.6b3q2qbjveo4zauc6xig7au10.my-network
+    Address 2: 10.0.9.3 my_web.1.63s86gf6a0ms34mvboniev7bs.my-network
+    Address 3: 10.0.9.5 my_web.3.66u2hcrz0miqpc8h0y0f3v7aw.my-network
+    ```
+
+6. Use `wget` from the busybox service to access the nginx web server running in
+the `my_web` service:
+
+    ```bash
+    # From inside the busybox conatiner
+    $ wget -O- my_web
+
+    Connecting to my_web (10.0.9.2:80)
+    <!DOCTYPE html>
+    <html>
+    <head>
+    <title>Welcome to nginx!</title>
+    ...snip...
+    ```
+
+    The swarm load balancer automatically routes the HTTP request to the service
+    VIP to an active task. It will distribute subsequent requests to other tasks
+    using round-robin selection.
+
+## Use DNS round-robin for a service
+
+You can configure the service to use DNS round-robin directly without using a
+VIP, by setting the `--endpoint-mode dnsrr` when you create the service. For example:
+
```

Comment:
Perhaps a short explanation why you would want to do this (DNS RR), e.g. to use your own load balancer, that needs to connect directly with the individual containers